src/bin/common/solving_loop.ml

```diff
@@ -95,6 +95,52 @@ let enable_maxsmt b =
   else
     DStd.Extensions.Smtlib2.(disable maxsmt)
 
+(* Dolmen util *)
+
+(** Returns the list of terms inside a dolmen binder. *)
+let terms_of_dolmen_binder = function
+  | DStd.Expr.Let_seq l | Let_par l ->
+    List.map snd l
+  | Lambda _ | Exists _ | Forall _ -> []
+
+(** Adds the named terms of the term [term] to the map accumulator [acc] *)
+let get_named_of_term
+    (acc : DStd.Expr.term Util.MS.t)
+    (term : DStd.Expr.term) =
+  let rec loop acc terms_to_check =
+    match terms_to_check with
+    | [] -> acc
+    | [] :: rest -> loop acc rest
+    | (term :: tl) :: rest ->
+      let terms_to_check = tl :: rest in
+      let terms_to_check =
+        match term.DStd.Expr.term_descr with
+        | DStd.Expr.Var _ | Cst _ -> terms_to_check
+        | App (t, _, tl) -> (t :: tl) :: terms_to_check
+        | Binder (b, t) ->
+          (t :: terms_of_dolmen_binder b) :: terms_to_check
```

Comment:
Is this going to do the right thing wrt capture of bound variables (e.g. what happens if I do `(let ((x 2)) (+ (! x :named name)))`)?

Reply:
I did not think about the semantics of named terms in bindings. Do you think we should not record the named formulas in bindings for now?

Reply:
If we don't have an easy way of checking capture avoidance then yes, that would be better (this is already what we do for quantifiers if I am not mistaken).